File: aula 3/com.br.gft/src/com/objetos/Caminhao.java
package com.objetos;

public class Caminhao extends Veiculo{
	private double tamanho;
	private String carga;
	public Caminhao(String marca, String modelo, String placa, float km, boolean isLigado, int litrosCombustivel,
			int velocidade, double preco, double tamanho, String carga) {
		super(marca, modelo, placa, km, isLigado, litrosCombustivel, velocidade, preco);
		this.tamanho = tamanho;
		this.carga = carga;
	}
	
	
	public double getTamanho() {
		return tamanho;
	}
	public void setTamanho(double tamanho) {
		this.tamanho = tamanho;
	}
	public String getCarga() {
		return carga;
	}
	public void setCarga(String carga) {
		this.carga = carga;
	}

	
	
	public void abastecer(float qtdLitros) {
		if (litrosCombustivel >= 150) {
			System.out.println("Combustivel cheio");
		} else {
			if (litrosCombustivel + qtdLitros >= 150) {
				System.out.println("abastecido 100%");
				litrosCombustivel = 150;
			} else {
				litrosCombustivel += qtdLitros;
			}

		}
	}
	
	@Override
	public void acelerar() {
		if (this.isLigado == true) {
			velocidade += 25;
			litrosCombustivel -= 5;
		} else if (this.litrosCombustivel < 1) {
			System.out.println("impossivel acelerar");
		} else {
			System.out.println("o veiculo está desligado");
		}
	}
	
}

File: C#/Veiculo/Veiculo/objetos/Caminhao.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Veiculo.objetos
{
    class Caminhao : Veiculo
    {
        private double tamanho;
        private String carga;

        public Caminhao(string marca, string modelo, string placa, float km, bool isLigado, int litrosCombustivel, int velocidade, double preco, double tamanho, String carga) : base(marca, modelo, placa, km, isLigado, litrosCombustivel, velocidade, preco)
        {
            this.Tamanho = tamanho;
            this.Carga = carga;
        }

		public double Tamanho { get => tamanho; set => tamanho = value; }
		public string Carga { get => carga; set => carga = value; }

		public void abastecer(float qtdLitros)
		{
			if (LitrosCombustivel >= 150)
			{
				Console.WriteLine("Combustivel cheio");
			}
			else
			{
				if (LitrosCombustivel + qtdLitros >= 150)
				{
					Console.WriteLine("abastecido 100%");
					LitrosCombustivel = 150;
				}
				else
				{
					LitrosCombustivel += (int) qtdLitros;
				}

			}
		}

	public void acelerar()
		{
			if (IsLigado == true)
			{
				Velocidade += 25;
				LitrosCombustivel -= 5;
			}
			else if (LitrosCombustivel < 1)
			{
				Console.WriteLine("impossivel acelerar");
			}
			else
			{
				Console.WriteLine("o veiculo está desligado");
			}
		}
	}
}
